base/parse.jl

```diff
@@ -194,3 +194,6 @@ function parse(str::AbstractString; raise::Bool=true)
     end
     return ex
 end
+
+promote_op{R<:AbstractFloat,S<:AbstractString}(::typeof(parse), ::Type{R}, ::Type{S}) = R
```

Comment:
This really starts looking like a reinvention of return type declarations. I think @JeffBezanson wouldn't like us to rely on the `return_type` function, but maybe we could make some exceptions? Anyway, working around that "prohibition" by inventing a parallel (and less practical) mechanism doesn't sound like the best strategy. ;-)